react/hooks/useStore.tsx

```diff
@@ -5,17 +5,23 @@ import { ShowAlertOptions } from '../utils/showAlertOptions'
 interface ContextStoreType {
   button: boolean
   setButton: (button: boolean) => void
-  valueGiftCard: number | undefined
+  valueGiftCard: number
   code: string
   addValueGiftCard: string | undefined
   setAddValueGiftCard: (addValueGiftCard: string) => void
   updateGiftCardFunction: () => void
   validation: string
   setValidation: (validation: string) => void
-  valueLists: number
   showAlert: ShowAlertOptions
+  setShowAlert: (showAlert: ShowAlertOptions) => void
   handleCloseAlert: () => void
   credit: number
+  copyCode: () => void
+  loading: boolean
+  loadingCode: boolean
+  rescue: number
+  borderRed: boolean
+  setBorderRed: (borderRed: boolean) => void
```

Comment:
This solution got better!
What do you think about the name of this property? Because in my opinion `borderRed` is very specific in this case.
Maybe a better name is `isGiftCardFieldValid`, or something like that 🤔

Reply:
Great idea, i will change where!